src/Microsoft.AspNet.Mvc.ModelBinding/Validation/AssociatedValidatorProvider.cs

```diff
@@ -38,15 +40,14 @@ private IEnumerable<IModelValidator> GetValidatorsForProperty(ModelMetadata meta
                     "metadata");
             }
 
-            var attributes = property.GetCustomAttributes();
-            return GetValidators(metadata, attributes);
+            var attributes = ModelAttributes.GetAttributesForProperty(metadata.ContainerType, property);
+            return GetValidators(metadata, attributes.Cast<Attribute>());
         }
 
         private IEnumerable<IModelValidator> GetValidatorsForType(ModelMetadata metadata)
         {
-            var attributes = metadata.ModelType
-                                     .GetTypeInfo()
-                                     .GetCustomAttributes();
+            var attributes = ModelAttributes.GetModelMetadataAttributesForType(metadata.ModelType);
+
             return GetValidators(metadata, attributes);
```

Comment:
do our overrides of `GetValidators()` return multiple validators when the attributes collection contains multiple attributes of the same `Type`?  may need to update `DataAnnotationsModelValidatorProvider` to avoid this problem, skipping to the next attribute if a `Type` has been seen.

(this issue likely wasn't possible before but can certainly happen with `[ModelMetadataType]`)